src/intra_edge.rs

```diff
@@ -324,17 +323,21 @@ impl<const SB128: bool, const N_BRANCH: usize, const N_TIP: usize>
 
         let sb128 = SB128 as u8;
 
-        let mut bl = BL_128X128;
-        while bl <= BL_32X32 {
-            indices.branch[bl as usize].index = level_index(bl + sb128);
+        let mut bl = BlockLevel::Bl128x128 as u8;
+        while bl <= BlockLevel::Bl32x32 as u8 {
```

Comment:
Can this be a `while let Some` with `.decrease`?

Reply:
I don't see how. upon entering the loop you immediately `.decrease` so it's impossible to cover `128x128` that way. We could add a `.increase`  that goes the opposite direction but I don't think that's worth it.

```
        let mut bl = BlockLevel::Bl16x16 ;
        while let Some(bl) = bl.increase() {
            indices.branch[bl as usize].index = level_index(bl as u8 + sb128);
        }
```

in particular it is kind of subtle that `16x16` is actually skipped here.

Reply:
Oh yeah, true, I didn't think about that.  Nevermind then.